Sources/SwiftDocCUtilities/CatalogTemplates/CatalogTemplateFactory.swift

```diff
@@ -0,0 +1,61 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021-2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import SwiftDocC
+
+protocol Catalog {
+    var articles: [String : ArticleTemplate] { get }
+    var title: String { get set }
+}
+
+struct CatalogTemplateFactory {
+    
+    enum Error: DescribedError {
+        case missingCatalogTemplate
+        var errorDescription: String {
+            switch self {
+                case .missingCatalogTemplate:
+                return "Provided template name is not a valid catalog template."
+            }
+        }
+    }
+    
+    let fileManager: FileManager = .default
+    
+    func createDocumentationCatalog(
+        _ catalogTemplate: CatalogTemplateKind,
+        catalogTitle: String
+    ) throws -> Catalog {
+        switch catalogTemplate {
+        case .base:
+            InitTemplateCatalog(title: catalogTitle)
+        }
+    }
+    
+    @discardableResult
+    func constructCatalog(
+        _ catalogTemplate: Catalog,
+        outputPath: String
+    ) throws -> URL {
+        try catalogTemplate.articles.forEach {
+            try fileManager.createDirectory(
+                at: URL(fileURLWithPath: "/\(outputPath)/\($0.key)").deletingLastPathComponent(),
```

Comment:
Instead of creating a URL with an extra path component and then deleting the last component you can avoid adding the extra path component.

```suggestion
                at: URL(fileURLWithPath: outputPath)
```

Reply:
The key is composed by the folder name + the file name. This logic is removing the file name from the key string 
https://github.com/apple/swift-docc/pull/719/files#diff-097599b64e594df0166fcaed57fb1295f9c8291c8b0c974f5e92de4afc932bd8R52

Reply:
In that case I think the key should be a URL to avoid using string manipulation to construct URLs.